File: rudder-sdk-core/src/main/java/com/rudderstack/android/sdk/core/RudderOSInfo.java
package com.rudderstack.android.sdk.core;

import android.os.Build;

import com.google.gson.annotations.SerializedName;

class RudderOSInfo {
    @SerializedName("name")
    private String name = "Android";
    @SerializedName("version")
    private String version = Build.VERSION.RELEASE;

     String getName() {
        return name;
    }

     String getVersion() {
        return version;
    }
}

File: models/src/main/java/com/rudderstack/models/android/RudderOSInfo.kt
package com.rudderstack.models.android

import com.google.gson.annotations.SerializedName

class RudderOSInfo(
    @SerializedName("name")
    private val name: String = "Android",

    @SerializedName("version") // = Build.VERSION.RELEASE
    private val version: String? = null,
)
